### ISSUE TITLE
  return glob.glob(os.path.join(gtav_dir, 'Installers', 'Social-Club*.exe'))[0] IndexError: list index out of range

### ISSUE THREAD

**Issue by monajalal** [openai/universe-windows-envs#10]
Can you please guide how to fix this error?
```

Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.

PS C:\Users\mona> cd $env:UNIVERSE_WINDOWS_ENVS_DIR\vnc-gtav
PS C:\Workspace\universe-windows-envs\vnc-gtav> python install.py
INFO:root:Downloading paths.xml file to C:\Program Files\Rockstar Games\Grand Theft Auto V - file is 150MB so this may t
ake a while
INFO:root:Downloading https://www.dropbox.com/s/6qc900o3a574yxc/paths.xml?dl=1
INFO:root:Backing up saved games in C:\Users\mona\Documents\Rockstar Games\GTA V\Profiles\ to C:\Users\mona\Documents\Un
iverse\GTAV_saved_games_backup_20170112_2130_38
INFO:root:Replacing saved games in C:\Users\mona\Documents\Rockstar Games\GTA V\Profiles\32E6C4B4
Verifying GTAV version...
Traceback (most recent call last):
  File "install.py", line 98, in <module>
    sys.exit(main())
  File "install.py", line 90, in main
    setup()

I see error is coming from here but I don't know how to fix it https://github.com/openai/universe-windows-envs/blob/f5aad9656661e8822c6dbe8dc212bf5f304a6c6e/vnc-gtav/enforce_version.py 
  File "install.py", line 48, in setup
    enforce_version(GTAV_DIR)
  File "C:\Workspace\universe-windows-envs\vnc-gtav\enforce_version.py", line 104, in enforce_version
    changes_detected = detect_binary_changes(gtav_dir)
  File "C:\Workspace\universe-windows-envs\vnc-gtav\enforce_version.py", line 28, in detect_binary_changes
    ('7129fb28c6a679f36b9c9461e1cadf6a', get_social_club_path(gtav_dir))
  File "C:\Workspace\universe-windows-envs\vnc-gtav\enforce_version.py", line 64, in get_social_club_path
    return glob.glob(os.path.join(gtav_dir, 'Installers', 'Social-Club*.exe'))[0]
IndexError: list index out of range
PS C:\Workspace\universe-windows-envs\vnc-gtav>
```
![kkk](https://cloud.githubusercontent.com/assets/1892917/21917880/ed49b604-d910-11e6-9736-759929484283.PNG)

**Comment by crizCraig** [comment]
thanks @monajalal - so this looks to be a problem with using the non-steam version of the game which doesn't have SocialClub in the `Installers` directory. @travelbureau, you're using this version correct? any tricks here?

**Comment by monajalal** [comment]
@travelbureau @crizCraig I don't have a Social-Club*.exe in my machine:
![ddf](https://cloud.githubusercontent.com/assets/1892917/21945804/fcb86ac6-d9a2-11e6-83ab-4a0eafbd7c00.png)
Also here's what in my Social  Club directory. Please share your hack with me too.
![socialclub](https://cloud.githubusercontent.com/assets/1892917/21945829/1ad98a62-d9a3-11e6-82f6-0f47969f8123.PNG)

**Comment by khooz** [comment]
Steam uses two main processes, `SteamService.exe` (the orchestrator) and [in case of games run by steam] `GameOverlayUI.exe` and `SteamClientService.exe`. The latter handles the game and DRM and the former handles in-game Steam overlay UI. Analysing processes, using Microsofts Process Explorer (aka. `procexp.exe`), seems that steam spawns `subprocess.exe` in the `$PROG_FILES\Rockstar Games\Social Club`.

**Comment by khooz** [comment]
I think you should first check for the existance of `steam_api.dll`. If it exists, look for `subprocess.exe`; otherwise look for `Social-Club*.exe`. All Steam integrated programs have `steam_api.dll`. You can read more on Steam integration here: https://partner.steamgames.com

**Comment by monajalal** [comment]
So I changed the enforce_version.py to this https://gist.github.com/monajalal/93cd3c40b4432d84e6cd0831d248c449 :
And here's the log:
```
PS C:\Workspace\universe-windows-envs\vnc-gtav> python install.py
INFO:root:Downloading paths.xml file to C:\Program Files\Rockstar Games\Grand Theft Auto V - file is 150MB so this may take a while
INFO:root:Downloading https://www.dropbox.com/s/6qc900o3a574yxc/paths.xml?dl=1
INFO:root:Backing up saved games in C:\Users\mona\Documents\Rockstar Games\GTA V\Profiles\ to C:\Users\mona\Documents\Universe\GTAV_saved_games_backup_20170113_1749_06
INFO:root:Replacing saved games in C:\Users\mona\Documents\Rockstar Games\GTA V\Profiles\32E6C4B4
Verifying GTAV version...

------------------------------------------------------------------------------------------------------------------------
FILE CHANGE DETECTED - C:\Program Files\Rockstar Games\Grand Theft Auto V\GTA5.exe
------------------------------------------------------------------------------------------------------------------------

Detected change in GTAV files. This likely means the game was updated which breaks ScriptHookV and can cause unwanted changes to environment dynamics.
Do you want to restore to a known working version of the game?
Note that this WILL NOT re-download the entire game or circumvent DRM - it will just roll back the updated portion of the game (y/n)? y
GTA5.exe is open, please close it and press any key to continue...

Downloading GTAV binaries - these are 665MB, so could take a while...
Extracting GTAV binaries from c:\users\mona\appdata\local\temp\tmpr2y1ym to C:\Program Files\Rockstar Games\Grand Theft Auto V
Finished extracting GTAV files, deleting temp file: c:\users\mona\appdata\local\temp\tmpr2y1ym
INFO:root:Backing up C:\Program Files\Rockstar Games\Grand Theft Auto V\dinput8.dll to C:\Program Files\Rockstar Games\Grand Theft Auto V\dinput8.dll.20170113_1753_05.backup
INFO:root:Copying C:\Workspace\universe-windows-envs\vnc-gtav\GTAVScriptHookProxy\bin\dinput8.dll to C:\Program Files\Rockstar Games\Grand Theft Auto V\dinput8.dll
INFO:root:Copying C:\Workspace\universe-windows-envs\vnc-gtav\GTAVScriptHookProxy\bin\GTAVScriptHookProxy.asi to C:\Program Files\Rockstar Games\Grand Theft Auto V\GTAVScriptHookProxy.asi
INFO:root:Copying C:\Workspace\universe-windows-envs\vnc-gtav\GTAVScriptHookProxy\bin\ScriptHookV.dev to C:\Program Files\Rockstar Games\Grand Theft Auto V\ScriptHookV.dev
INFO:root:Copying C:\Workspace\universe-windows-envs\vnc-gtav\GTAVScriptHookProxy\bin\ScriptHookV.dll to C:\Program Files\Rockstar Games\Grand Theft Auto V\ScriptHookV.dll
INFO:root:Copying C:\Workspace\universe-windows-envs\vnc-gtav\GTAVScriptHookProxy\bin\x360ce_x64.exe to C:\Program Files\Rockstar Games\Grand Theft Auto V\x360ce_x64.exe
INFO:root:Installation complete

```
Seems it is working but I will close this issue when I am totally sure commenting those lines won't hurt.

**Comment by wear-your-seatbelt** [comment]
@monajalal - the enforce_version script is following advice on ScriptHookV forums to restore an older version of GTAV by copying three files, including SocialClub. SocialClub is used across Rockstar games and so it's probably fine to have a new version of that and an old version of the GTA executables `GTAVLauncher.exe` and `GTA5.exe`. That being said, the script should be updated to use the correct location of SocialClub which should be here:

```
HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Rockstar Games\Rockstar Games Social Club
```

**Comment by wear-your-seatbelt** [comment]
This will be fixed with #10
